browser_patches/firefox-beta/patches/bootstrap.diff

```diff
@@ -171,6 +364,67 @@ index 040c7b124dec6bb254563bbe74fe50012cb077a3..b4e6b8132786af70e8ad0dce88b67c28
  
    const transportProvider = {
      setListener(upgradeListener) {
+diff --git a/devtools/startup/DevToolsStartup.jsm b/devtools/startup/DevToolsStartup.jsm
+index 2c1f767f89f596fb5732572ad4e0cad3cafcc7c3..99401071ad93a2f2a3ba37e7643920f2e05e8b25 100644
+--- a/devtools/startup/DevToolsStartup.jsm
++++ b/devtools/startup/DevToolsStartup.jsm
+@@ -375,7 +375,7 @@ DevToolsStartup.prototype = {
+ 
+       // Store devtoolsFlag to check it later in onWindowReady.
+       this.devtoolsFlag = flags.devtools;
+-
++      this.autoOpenDevtoolsForTabs = flags.autoOpenDevtoolsForTabs;
+       /* eslint-disable mozilla/balanced-observers */
+       // We are not expecting to remove those listeners until Firefox closes.
+ 
+@@ -438,6 +438,7 @@ DevToolsStartup.prototype = {
+ 
+     const console = cmdLine.handleFlag("jsconsole", false);
+     const devtools = cmdLine.handleFlag("devtools", false);
++    const autoOpenDevtoolsForTabs = cmdLine.handleFlag("auto-open-devtools-for-tabs", false);
+ 
+     let devToolsServer;
+     try {
+@@ -460,7 +461,7 @@ DevToolsStartup.prototype = {
+       debuggerFlag = cmdLine.handleFlag("jsdebugger", false);
+     }
+ 
+-    return { console, debugger: debuggerFlag, devtools, devToolsServer };
++    return { console, debugger: debuggerFlag, devtools, autoOpenDevtoolsForTabs, devToolsServer };
+   },
+ 
+   /**
+@@ -482,9 +483,22 @@ DevToolsStartup.prototype = {
+       this._firstWindowReadyReceived = true;
+     }
+ 
++    if (this.autoOpenDevtoolsForTabs) {
++      this.handleDevToolsOpenForEveryTab(window);
++    }
++
+     JsonView.initialize();
+   },
+ 
++  handleDevToolsOpenForEveryTab(window) {
++    const require = this.initDevTools("CommandLine");
++    const { gDevTools } = require("devtools/client/framework/devtools");
++    window.gBrowser.tabContainer.addEventListener('TabOpen', async (event) => {
++      await gDevTools.showToolboxForTab(event.target);
++    });
++    gDevTools.showToolboxForTab(window.gBrowser.selectedTab).catch(() => {});
```

Comment:
I assume DevTools have their own browser and this code will not run recursively.

Reply:
Currently this runs in the Playwright world per Page, since we create for every Page a new Window.

At least from my testing it didn't.